crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,262 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+
+use biome_js_syntax::JsStringLiteralExpression;
+
+use biome_rowan::AstNode;
+use regex::Regex;
+
+use std::sync::{Arc, LazyLock, Mutex};
+use std::thread;
+
+enum Pattern {
+    Regex(&'static LazyLock<Regex>),
+    Contains(&'static str),
+}
+
+// TODO: Try to get this to work in JavaScript comments as well
+declare_lint_rule! {
+    /// Disallow usage of sensitive data such as API keys and tokens.
+    ///
+    /// This rule checks for high-entropy strings and matches common patterns
+    /// for secrets, such as AWS keys, Slack tokens, and private keys.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var secret = "AKIA1234567890EXAMPLE";
```

Comment:
Not a huge thing, but can we use newer ESM syntax in code examples? `const`

Reply:
Done, thanks.

Reply:
I meant in documentation, snapshots can use whatever you want, but user visible information should guide users to use right things. Use of var is discouraged, we even have rule banning it https://biomejs.dev/linter/rules/no-var/

Reply:
Oh sorry! Nice catch.